### PR TITLE
Revise reduce mean

### PR DESCRIPTION
**Pull request by pszmel**
1. Update spec
2. Add type_prop tests

ticket: 37468

### CODE REVIEW

File: docs/ops/reduction/ReduceMean_1.md [hunk 1]
```diff
@@ -33,11 +33,11 @@
 
 **Detailed Description**
 
-Each element in the output is the result of reduction with finding the arithmetic mean operation along dimensions specified by the 2nd input:
+Each element in the output is the result of arithmetic mean reduction operation along dimensions specified by the 2nd input:
 
     output[i0, i1, ..., iN] = mean[j0,..., jN](x[j0, ..., jN]))
 
-Where indices i0, ..., iN run through all valid indices for the 1st input and finding the arithmetic mean `mean[j0, ..., jN]` have `jk = ik` for those dimensions `k` that are not in the set of indices specified by the 2nd input of the operation. 
+Where indices i0, ..., iN run through all valid indices for the 1st input and finding the arithmetic mean `mean[j0, ..., jN]`.`jk = ik` for those dimensions `k` that are not in the set of indices specified by the 2nd input of the operation.
```
Comment (ilyachur) [inline docs/ops/reduction/ReduceMean_1.md:40]: Why did you change `have` to `.`?

Comment (pszmel) [reply docs/ops/reduction/ReduceMean_1.md:40]: I misunderstood the sentence. I will undo it

File: ngraph/test/type_prop/reduce_mean.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{3, 4, 5});
+    auto axes = make_shared<op::Constant>(element::i64, Shape{2}, vector<int64_t>{1, 2});
+    auto keep_dims = true;
+    auto reduce_prod = make_shared<op::v1::ReduceMean>(arg, axes, keep_dims);
+    ASSERT_TRUE(reduce_prod->get_output_partial_shape(0).compatible(PartialShape{3, 1, 1}));
+}
+
+TEST(type_prop, reduce_mean_v1_shape_if_not_keep_dims)
+{
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{3, 4, 5});
```
Comment (ilyachur) [inline ngraph/test/type_prop/reduce_mean.cpp:56]: Can we add tests for dynamic shapes?

**Review by ilyachur — APPROVED** [review]

**Review by pelszkow — APPROVED** [review]
